src/dsp_tools/utils/shared.py

```diff
@@ -162,11 +172,11 @@ def prepare_dataframe(df: pd.DataFrame, required_columns: list[str], location_of
     # delete rows that don't have the required columns
     for req in required_columns:
         if req not in new_df:
-            raise ValueError(f"{location_of_sheet} requires a column named '{req}'")
+            raise BaseError(f"{location_of_sheet} requires a column named '{req}'")
```

Comment:
Is the idea to only use BaseError in the future? Or do you plan to have other Errors as well?